### PR TITLE
MV-449 Remove apex amp option

### PR DESCRIPTION
**Pull request by loctxmoreh**
## Background 
[MV-416](https://moreh.atlassian.net/browse/MV-416) & [MV-449](https://moreh.atlassian.net/browse/MV-449)

## Contents
* Remove the use of `torch.moreh.apex`
* Remove the option to specify apex optimization level 

## Note
This PRs should only be merged after configuration files in Airflow are updated to remove the use of `amp_opt_level`

[MV-416]: https://moreh.atlassian.net/browse/MV-416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MV-449]: https://moreh.atlassian.net/browse/MV-449?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -359,8 +345,6 @@
                    help='lower precision AMP dtype (default: float16)')
 group.add_argument('--amp-impl', default='native', type=str,
                    help='AMP impl to use, "native" or "apex" (default: native)')
-group.add_argument('--amp-opt-level', default='O1', type=str,
```
Comment (dohyung-kim-moreh) [inline train.py:362]: https://github.com/moreh-dev/moreh-testbench/blob/main/yamls/nexus/tiny_set_acc_without_issues.yaml#L810

We need to remove O2 tests in our yamls.

File: train.py [hunk 2]
```diff
@@ -44,18 +44,7 @@
     from apex.parallel import convert_syncbn_model
     has_apex = True
 except ImportError:
-    if not hasattr(torch, 'moreh'):
```
Comment (dohyung-kim-moreh) [inline train.py:47]: If we remove amp tests, then we don't need amp tests of NVIDIA GPU as well. 
It means we can remove out all amp tests in our model config yamls?

File: train.py (hunk 1 above)
Comment (dohyung-kim-moreh) [reply train.py:362]: Ah I just saw [this PR](https://github.com/moreh-dev/moreh-testbench/pull/197).

In MAF, is it available  pytorch native amp? 
```
    amp_impl: native
    amp_dtype: float16
```

File: train.py (hunk 2 above)
Comment (dohyung-kim-moreh) [reply train.py:47]: Ah I just saw https://github.com/moreh-dev/moreh-testbench/pull/197.

File: train.py (hunk 1 above)
Comment (loctxmoreh) [reply train.py:362]: > In MAF, is it available pytorch native amp?

Yes it is

**Comment by loctxmoreh** [comment]
@yeomni @dohyung-kim-moreh Please approve this if there is no problem left

**Review by yeomni — APPROVED** [review]
I thought I've already approved this.. my bad. LGTM

**Pull request merged**